### ISSUE TITLE
The `@each` function does not inherit variables from main view, while `@include` does so

### ISSUE THREAD

**Issue by UnrulyNatives** [laravel/framework#13209]
In my view file I have this loop

    @foreach($object as $o)
        @include('displaymodules._showmodule_'.$showmodule_type)
    @endforeach

```
it works perfectly well, but... I wanted to make my code more beautiful and use `@each` function

https://laravel.com/docs/5.2/blade#displaying-data

`@each('displaymodules._showmodule_'.$itemkind.'_'.$showmodule_type, $option, 'o', 'segments._notice_no_data')`

unfortunately, after I implemented the rewrite, the variable `$object` is not recognized within the template, which full name (including strings provided by varaibles) is `displaymodules._itemtypevalue_dimmer`. It seems that variables (in case pf `$itemkind` it is defined in my controller) are not inherited by the view defined in `@each`. I get this error

    Undefined variable: itemkind (View: \resources\views\displaymodules\_showmodule_itemtypevalue_dimmer.blade.php)


In other words, the variable works in the main view and it also works in the original solution's sub-view defined by `       @include('displaymodules._showmodule_'.$showmodule_type)`

------------------

Please also refer to my question in StackOverflow, which is pasted above.
http://stackoverflow.com/questions/36685456/the-each-function-does-not-inherit-variables-from-main-view-while-include

**Comment by mark86092** [comment]
See #13090

**Comment by robclancy** [comment]
> I wanted to make my code more beautiful and use `@each` function

proceeds to make the code look terrible
